e2e/main.py

```diff
@@ -543,14 +548,7 @@ def get_collateral_tokens_and_make_checker_an_operator(amnt):
             "callback": None,
         }
         print(f"Calling balance_of as an off-chain view with {fa2_balance_of}")
-        balance = checker.balance_of(**fa2_balance_of).callback_view()
-        # Check that this balance agrees with the kit balance view
-        assert_fa2_token_balance(
-            checker,
-            account_alice,
-            kit_token_id,
-            balance[0]["nat_2"],
-        )
+        checker.balance_of(**fa2_balance_of).callback_view()
```

Comment:
Shouldn't we also check the result of `balance_of` here?

Reply:
Ah, so it turns out that we actually call `balance_of` in the implementation of `assert_fa2_token_balance`, so this check was actually just comparing the same value :upside_down_face:

Reply:
Haha, nice :stuck_out_tongue:

Still though, since we call
```python
        call_checker_endpoint("transfer", [fa2_transfer])
        assert_fa2_token_balance(checker, account_alice, kit_token_id, 10)
```
a few lines above, this whole bit
```python
        # `balance_of` requires a contract callback when executing on-chain. To make tests
        # more light-weight and avoid needing an additional mock contract, we call it as a view.
        # This comes with a downside, however, since using a view means that we don't get an
        # estimate of the gas cost.
        fa2_balance_of = {
            "requests": [{"owner": account_alice, "token_id": kit_token_id}],
            "callback": None,
        }
        print(f"Calling balance_of as an off-chain view with {fa2_balance_of}")
        checker.balance_of(**fa2_balance_of).callback_view()
```
could go, no? I mean, I don't see any benefit in calling `checker.balance_of(**fa2_balance_of).callback_view()` here

Reply:
Very true. IIRC this ended up this way since we had a different implementation of `assert_fa2_token_balance` originally. I guess the only argument for keeping it there would be in-case we change the implementation of `assert_fa2_token_balance` in the future, though a comment to this end would probably also suffice.